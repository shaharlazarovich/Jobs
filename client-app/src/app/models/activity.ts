export interface IAttendee {
    userName: string;
    displayName: string;
    image: string;
    isHost: boolean;
    following?: boolean,
}

//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our activity array, in order to add a count
//of activities - which then, we could device in the number of pages
export interface IActivitiesEnvelope {
    activities: IActivity[];
    activityCount: number
}

export interface IActivity {
    id: string;
    title: string;
    description: string;
    category: string;
    date: Date;
    city: string;
    venue: string;
    isGoing: boolean;
    isHost: boolean;
    attendees: IAttendee[];
    comments: IComment[];
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
}

export interface IActivityFormValues extends Partial<IActivity> {
    time?: Date    
}

export class ActivityFormValues implements IActivityFormValues {
    id?: string = undefined;
    title: string = '';
    category: string = '';
    description: string = '';
    date?:Date = undefined;
    time?: Date = undefined;
    city: string = '';
    venue: string = '';

    constructor(init?: IActivityFormValues) {
        if (init && init.date) {
            init.time = init.date;
        }
        Object.assign(this, init); //this will initialize our object of class ActivityFormValues
                                  // with the above values
    }
}