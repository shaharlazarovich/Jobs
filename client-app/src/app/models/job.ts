//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface IJobsEnvelope {
    jobs: IJob[];
    jobsCount: number
}

export interface IJob {
    id: number;
    jobName: string;
    company: string;
    replication: string;
    servers: string;
    lastRun: Date;
    date: Date;
    time: Date;
    rta: string;
    results: string;
    key: string;
    rtoNeeded: string;
    jobIP: string;
}


export interface IJobFormValues extends Partial<IJob> {
     
}

export class JobFormValues implements IJobFormValues {
    id?: number = 0;
    jobName: string = '';
    company: string = '';
    replication: string = '';
    servers: string = '';
    lastRun?:Date = undefined;
    date?:Date = undefined;
    time?:Date = undefined;
    rta?: string = '';
    results: string = '';
    key: string = '';
    rtoNeeded: string = '';
    jobIP: string = '';

    constructor(init?: IJobFormValues) {
        Object.assign(this, init); //this will initialize our object of class JobFormValues
                                  // with the above values
    }
}