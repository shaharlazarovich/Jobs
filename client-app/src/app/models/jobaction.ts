//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface IJobActionsEnvelope {
    jobActions: IJobAction[];
    jobActionsCount: number
}

export interface IJobAction {
    id: string;
    jobId: string;
    jobName: string;
    userId: string;
    actionDate: Date;
    remoteIP: string;
    remoteResponse: string;
    requestProperties: string;
    source: string;
    action: string;
}

export class JobAction implements IJobAction {
    id: string = '';
    jobId: string = '';
    jobName: string = '';
    userId: string = '';
    actionDate: Date = new Date();
    remoteIP: string = '';
    remoteResponse: string = '';
    requestProperties: string = '';
    source: string = '';
    action: string = '';

    constructor(init?: IJobAction) {
        Object.assign(this, init); //this will initialize our object of class JobActions
                                  // with the above values
    }
}