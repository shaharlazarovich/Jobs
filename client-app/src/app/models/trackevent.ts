//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface ITrackEventsEnvelope {
    trackEvents: ITrackEvent[];
    trackEventsCount: number
}

export interface ITrackEvent {
    id: string;
    jobId?: string;
    jobName?: string;
    userId?: string;
    actionDate: Date;
    remoteIP?: string;
    remoteResponse?: string;
    requestProperties?: string;
    source: string;
    event: string;
}

export class TrackEvent implements ITrackEvent {
    id: string = '';
    jobId?: string = '';
    jobName?: string = '';
    userId?: string = '';
    actionDate: Date = new Date();
    remoteIP?: string = '';
    remoteResponse?: string = '';
    requestProperties?: string = '';
    source: string = '';
    event: string = '';

    constructor(init?: ITrackEvent) {
        Object.assign(this, init); //this will initialize our object of class JobActions
                                  // with the above values
    }
}