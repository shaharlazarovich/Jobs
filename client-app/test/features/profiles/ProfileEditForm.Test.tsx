import { shallow } from "enzyme";
import React from "react";
import { IProfile } from '../../../src/app/models/profile'
import { findByTestAttr } from '../../common/testUtils';
import ProfileEditForm from "../../../src/features/profiles/ProfileEditForm";

interface IProps {
  profile: IProfile;
}

const setup = (profile:any) => {
    const wrapper = shallow<IProps>(<ProfileEditForm {...profile} />)
    return wrapper;
}

describe('Profile Edit Form UniTest', () => {
    let wrapper:any;
    beforeEach(() => {
        wrapper = setup([]);
    });
    test('renders without error', () => {
        const component = findByTestAttr(wrapper, 'component-profile-edit-form');
        expect(component.length).toBe(1);
    });
})