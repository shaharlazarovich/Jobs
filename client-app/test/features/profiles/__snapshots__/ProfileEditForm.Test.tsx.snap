// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Edit Form UniTest should match the snapshot 1`] = `
<ReactFinalForm
  data-test="component-profile-edit-form"
  initialValues={
    Object {
      "bio": "fakeBIO",
      "displayName": "fakeName",
      "followersCount": 1,
      "following": true,
      "followingCount": 1,
      "image": "",
      "photos": null,
      "userName": "fakeUserName",
    }
  }
  onSubmit={[Function]}
  render={[Function]}
  validate={[Function]}
>
  <Form
    as="form"
    error={true}
    onSubmit={[Function]}
  >
    <form
      className="ui error form"
      onSubmit={[Function]}
    >
      <Field
        component={[Function]}
        name="displayName"
        placeholder="Display Name"
        value="fakeName"
      >
        <TextInput
          input={
            Object {
              "name": "displayName",
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "type": undefined,
              "value": "fakeName",
            }
          }
          meta={
            Object {
              "active": false,
              "data": Object {},
              "dirty": false,
              "dirtySinceLastSubmit": false,
              "error": undefined,
              "initial": "fakeName",
              "invalid": false,
              "length": undefined,
              "modified": false,
              "pristine": true,
              "submitError": undefined,
              "submitFailed": false,
              "submitSucceeded": false,
              "submitting": false,
              "touched": false,
              "valid": true,
              "validating": false,
              "visited": false,
            }
          }
          placeholder="Display Name"
        >
          <FormField
            error={false}
          >
            <div
              className="field"
            >
              <input
                name="displayName"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Display Name"
                value="fakeName"
              />
            </div>
          </FormField>
        </TextInput>
      </Field>
      <Field
        component={[Function]}
        name="bio"
        placeholder="Bio"
        rows={3}
        value="fakeBIO"
      >
        <TextAreaInput
          input={
            Object {
              "name": "bio",
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "type": undefined,
              "value": "fakeBIO",
            }
          }
          meta={
            Object {
              "active": false,
              "data": Object {},
              "dirty": false,
              "dirtySinceLastSubmit": false,
              "error": undefined,
              "initial": "fakeBIO",
              "invalid": false,
              "length": undefined,
              "modified": false,
              "pristine": true,
              "submitError": undefined,
              "submitFailed": false,
              "submitSucceeded": false,
              "submitting": false,
              "touched": false,
              "valid": true,
              "validating": false,
              "visited": false,
            }
          }
          placeholder="Bio"
          rows={3}
        >
          <FormField
            error={false}
          >
            <div
              className="field"
            >
              <textarea
                name="bio"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Bio"
                rows={3}
                value="fakeBIO"
              />
            </div>
          </FormField>
        </TextAreaInput>
      </Field>
      <Button
        as="button"
        content="Update profile"
        disabled={true}
        floated="right"
        loading={false}
        positive={true}
      >
        <Ref
          innerRef={
            Object {
              "current": <button
                class="ui positive disabled right floated button"
                disabled=""
                tabindex="-1"
              >
                Update profile
              </button>,
            }
          }
        >
          <RefFindNode
            innerRef={
              Object {
                "current": <button
                  class="ui positive disabled right floated button"
                  disabled=""
                  tabindex="-1"
                >
                  Update profile
                </button>,
              }
            }
          >
            <button
              className="ui positive disabled right floated button"
              disabled={true}
              onClick={[Function]}
              tabIndex={-1}
            >
              Update profile
            </button>
          </RefFindNode>
        </Ref>
      </Button>
    </form>
  </Form>
</ReactFinalForm>
`;
